pkg/api/v1alpha1/cluster_test.go

```diff
@@ -878,6 +878,12 @@ func TestGetAndValidateClusterConfig(t *testing.T) {
 			wantCluster: nil,
 			wantErr:     true,
 		},
+		{
+			testName:    "no control plane endpoint",
+			fileName:    "testdata/cluster_invalid_no_cp_endpoint.yaml",
```

Comment:
Could we test it without going through yaml? I understand other tests do it that way, but it's unnecessary complexity